pkg/ring/model.go

```diff
@@ -649,3 +649,105 @@ func MergeTokensByZone(zones map[string][][]uint32) map[string][]uint32 {
 	}
 	return out
 }
+
+func (d *Desc) SplitByID() map[string]interface{} {
+	out := make(map[string]interface{}, len(d.Ingesters))
+	for key := range d.Ingesters {
+		in := d.Ingesters[key]
+		out[key] = &in
+	}
+	return out
+}
+
+func (d *Desc) JoinIds(in map[string]interface{}) {
+	for key, value := range in {
+		d.Ingesters[key] = *(value.(*InstanceDesc))
+	}
+}
+
+func (d *Desc) GetItemFactory() proto.Message {
+	return &InstanceDesc{}
+}
+
+func (d *Desc) RefreshTimestamp() {
+	var fresherTimestamp int64
+	for _, ing := range d.Ingesters {
+		if ing.Timestamp > fresherTimestamp {
+			fresherTimestamp = ing.Timestamp
+		}
+	}
+	staleDuration := time.Now().UTC().Unix() - fresherTimestamp
+	for id, ing := range d.Ingesters {
+		ing.Timestamp += staleDuration
```

Comment:
We are shifting all the ingester's timestamp by the same amount here. I'm not sure why we can do this. Could we add some explanation to the function?

Reply:
Ah this is an old method from another experiment. I am not using anymore. I will remove the code